tests/test_nhg.py

```diff
@@ -216,6 +286,358 @@ def test_route_nhg(self, dvs, dvs_route, testlog):
+        fvs = swsscommon.FieldValuePairs([('nexthop_group', 'group2')])
+        rt_ps.set('2.2.2.0/24', fvs)
+
+        # The first group should have got deleted
+        asic_db.wait_for_n_keys(self.ASIC_NHG_STR, asic_nhgs_count + 1)
+        asic_db.wait_for_n_keys(self.ASIC_NHGM_STR, asic_nhgms_count + 2)
+
+        assert asic_db.get_entry(self.ASIC_RT_STR, rt_key)['SAI_ROUTE_ENTRY_ATTR_NEXT_HOP_ID'] != nhgid
+
+        # Update the route with routeOrch's owned next hop group
+        nhgid = asic_db.get_entry(self.ASIC_RT_STR, rt_key)['SAI_ROUTE_ENTRY_ATTR_NEXT_HOP_ID']
+        fvs = swsscommon.FieldValuePairs([('nexthop', '10.0.0.1,10.0.0.3'),
+                                          ('ifname', 'Ethernet0,Ethernet4')])
+        rt_ps.set('2.2.2.0/24', fvs)
+
+        asic_db.wait_for_n_keys(self.ASIC_NHG_STR, asic_nhgs_count + 2)
+        asic_db.wait_for_n_keys(self.ASIC_NHGM_STR, asic_nhgms_count + 4)
+
+        # Assert the next hop group ID changed
+        time.sleep(1)
+        assert asic_db.get_entry(self.ASIC_RT_STR, rt_key)['SAI_ROUTE_ENTRY_ATTR_NEXT_HOP_ID'] != nhgid
+        nhgid = asic_db.get_entry(self.ASIC_RT_STR, rt_key)['SAI_ROUTE_ENTRY_ATTR_NEXT_HOP_ID']
+
+        # Update the route to point back to group2
+        fvs = swsscommon.FieldValuePairs([('nexthop_group', 'group2')])
+        rt_ps.set('2.2.2.0/24', fvs)
+
+        # The routeOrch's owned next hop group should get deleted
+        asic_db.wait_for_n_keys(self.ASIC_NHG_STR, asic_nhgs_count + 1)
+        asic_db.wait_for_n_keys(self.ASIC_NHGM_STR, asic_nhgms_count + 2)
+
+        # Assert the route points back to group2
+        assert asic_db.get_entry(self.ASIC_RT_STR, rt_key)['SAI_ROUTE_ENTRY_ATTR_NEXT_HOP_ID'] != nhgid
+
+        # Create a new group with the same members as group2
+        nhgid = asic_db.get_entry(self.ASIC_RT_STR, rt_key)['SAI_ROUTE_ENTRY_ATTR_NEXT_HOP_ID']
+        fvs = swsscommon.FieldValuePairs([('nexthop', '10.0.0.1,10.0.0.3'),
+                                          ('ifname', 'Ethernet0,Ethernet4')])
+        nhg_ps.set("group1", fvs)
+        asic_db.wait_for_n_keys(self.ASIC_NHG_STR, asic_nhgs_count + 2)
+        asic_db.wait_for_n_keys(self.ASIC_NHGM_STR, asic_nhgms_count + 4)
+
+        # Update the route to point to the new group
+        fvs = swsscommon.FieldValuePairs([('nexthop_group', 'group1')])
+        rt_ps.set('2.2.2.0/24', fvs)
+        time.sleep(1)
+
+        # Assert the next hop group ID changed
+        assert asic_db.get_entry(self.ASIC_RT_STR, rt_key)['SAI_ROUTE_ENTRY_ATTR_NEXT_HOP_ID'] != nhgid
+
+        # Remove the route
+        rt_ps._del('2.2.2.0/24')
+        asic_db.wait_for_n_keys(self.ASIC_RT_STR, asic_rts_count)
+
+        # Remove the groups
+        nhg_ps._del('group1')
+        nhg_ps._del('group2')
+
+        asic_db.wait_for_n_keys(self.ASIC_NHG_STR, asic_nhgs_count)
+        asic_db.wait_for_n_keys(self.ASIC_NHGM_STR, asic_nhgms_count)
+
+        # Create a route with labeled NHs
```

Comment:
This can be a different test case. Advantage of small, specific test cases is it becomes easier to debug when there's a failure. But this is fine too.